supabase/migrations/35_prune_unchanged_draft_specs.sql

```diff
@@ -0,0 +1,141 @@
+
+begin;
+
+alter table inferred_schemas add column "md5" text generated always as (md5(trim("schema"::text))) stored;
+comment on column inferred_schemas.md5 is
+  'hash of the inferred schema json, which is used to identify changes';
+
+alter table live_specs add column inferred_schema_md5 text;
+comment on column live_specs.inferred_schema_md5 is
+  'The md5 sum of the inferred schema that was published with this spec';
+
+-- Re-define the live_specs_ext view to include the inferred_schema_md5 column,
+-- which will be needed by the unchanged_draft_specs view. Drop and re-create
+-- the views because selecting * changes the order of some columns.
+drop view draft_specs_ext;
+drop view live_specs_ext;
+
+-- l.* expands to an additional column now, but columns are otherwise identical to the previous view definition
+-- Extended view of live catalog specifications.
+create view live_specs_ext as
+select
+  l.*,
+  c.external_url as connector_external_url,
+  c.id as connector_id,
+  c.title as connector_title,
+  c.short_description as connector_short_description,
+  c.logo_url as connector_logo_url,
+  c.recommended as connector_recommended,
+  t.id as connector_tag_id,
+  t.documentation_url as connector_tag_documentation_url,
+  p.detail as last_pub_detail,
+  p.user_id as last_pub_user_id,
+  u.avatar_url as last_pub_user_avatar_url,
+  u.email as last_pub_user_email,
+  u.full_name as last_pub_user_full_name
+from live_specs l
+left outer join publication_specs p on l.id = p.live_spec_id and l.last_pub_id = p.pub_id
+left outer join connectors c on c.image_name = l.connector_image_name
+left outer join connector_tags t on c.id = t.connector_id and l.connector_image_tag = t.image_tag
+left outer join internal.user_profiles u on u.user_id = p.user_id
+-- This first condition allows superusers to query the view. The second is the normal RLS policy,
+-- but implemented here in a way that is more efficient when querying for large sets of specs.
+where exists(select 1 from pg_roles where rolname = current_role and rolbypassrls = true)
+  or l.id in (
+    -- User must be able to read catalog_name. Compare to select RLS policy.
+    select l.id from auth_roles('read') r, live_specs l
+      where l.catalog_name ^@ r.role_prefix
+  )
+;
+-- live_specs_ext includes its own authorization checks.
+grant select on live_specs_ext to authenticated;
+
+comment on view live_specs_ext is
+  'View of `live_specs` extended with metadata of its last publication';
+
+-- Extended view of user draft specifications.
+create view draft_specs_ext  as
+select
+  d.*,
+  l.last_pub_detail,
+  l.last_pub_id,
+  l.last_pub_user_id,
+  l.last_pub_user_avatar_url,
+  l.last_pub_user_email,
+  l.last_pub_user_full_name,
+  l.spec as live_spec,
+  l.spec_type as live_spec_type,
+  -- new columns below
+  s.md5 as inferred_schema_md5,
+  l.inferred_schema_md5 as effective_inferred_schema_md5,
```

Comment:
can we call this `live_infered_schema_md5` ? It wasn't clear to me what "effective" meant in this context. I mean, it was, but I spent brain cycles trying to figure out if it was something _different_ from the value of `live_specs`. We have existing precedent of `live_spec`, `live_spec_type`, `live_spec_md5` etc in this view.

Reply:
that's fair. I was trying to avoid ambiguity with "live" referring to the current value of `inferred_schemas.md5`, but I may be the only one who found that confusing.